R/print.R

```diff
@@ -75,6 +75,9 @@ print.card <- function(x, n = NULL, columns = c("auto", "all"), n_col = 6L, ...)
     x_print$statistic <- lapply(
       x_print$statistic,
       function(x) {
+        if (inherits(x, c("Date", "POSIXct", "POSIXt"))) {
```

Comment:
![image](https://github.com/insightsengineering/cards/assets/43018762/341de976-ac13-428f-845b-180d5c805aae)

Was receiving this error message if I tried to simply print the dataframe, this modification allows it simply work

Reply:
ahh, i see. i'll fix the print method now

Reply:
Well I think I "fixed" it, more of a questino whether or not `.default_fmt_fn()` still needs to be modified?

Reply:
The raw statistic is different from the formatted statistic, so we definitely still need the changes in `.default_fmt_fn()`.

I am a little unclear why the print method was error before your update.....let's jump on a call and chat.

Reply:
I am wondering if we should generalize our approach. Perhaps something like, if it's an integer, round to the nearest integer, if it's a percentage, scale by 100 and round, if it's double then round to 1 decimal place. Otherwise, `as.character()`.

Then we don't need special handing for new classes as they begin to pop up. What do you think?